cnd/src/swap_protocols/ledger_states.rs

```diff
@@ -0,0 +1,126 @@
+use crate::swap_protocols::{
+    rfc003::{create_swap::SwapEvent, LedgerState},
+    state::{Get, Insert, Update},
+    swap_id::SwapId,
+};
+use std::{
+    any::Any,
+    collections::HashMap,
+    ops::{Deref, DerefMut},
+    sync::Mutex,
```

Comment:
This is the wrong `Mutex`.  We need to use an async mutex, in other places we use `tokio::sync::Mutex` (see `db/mod.rs`).  Ref: https://rust-lang.github.io/async-book/03_async_await/01_chapter.html (search for 'Mutex').

Reply:
Sorry, auto-import!

Reply:
You will need to make these `async_trait`s with the tokio mutex.